src/app/model/User.ts

```diff
@@ -45,246 +43,72 @@ class User extends BaseModel {
-    }
-    try {
-      collection.doc(missionId).update({
-        status: MissionStatus.delivered,
-      });
-    } catch (e) {
-      //TODO show error msg to user
-      throw e;
-    }
+  createProfile(userUid: string, data: object) {
+    return this.getCollection("users")
+      .doc(userUid)
+      .set(this.load({ data, uid: userUid }));
   }
 
   /**
    * Returns all missions that a volunteer is associated with or has been suggested for.
-   * @param userId UserId of the volunteer
+   * @param userUid UserId of the volunteer
    */
-  async getAllAssociatedMissions(userId: string) {
-    const collection = this.getCollection("organizations").doc("1").collection("missions");
-
-    const volunteeredMissions = await collection.where("volunteerId", "==", userId).get();
-    const suggestedMissions = await collection.where("tentativeVolunteerId", "==", userId).get();
-
-    const missionsDocumentSnapshot = volunteeredMissions.docs.concat(suggestedMissions.docs);
-
-    if (missionsDocumentSnapshot.length < 1) {
-      return [];
-    }
-    const missions = missionsDocumentSnapshot.map((doc) => doc.data());
-
-    return missions;
+  getAllAssociatedMissions(userUid: string) {
+    const collection = this.getCollection("organizations")
+      .doc(Organization.uid)
+      .collection("missions");
+
+    return Promise.all([
+      collection
+        .where("volunteerUid", "==", userUid)
+        .where("status", "in", [MissionStatus.assigned, MissionStatus.started])
+        .get(),
+      collection
+        .where("tentativeVolunteerUid", "==", userUid)
+        .where("status", "in", [MissionStatus.tentative])
+        .get(),
+    ]).then(([volunteer, tentative]) => {
+      const docs = volunteer.docs.concat(tentative.docs);
+      return docs.map((doc) => doc.data());
+    });
   }
 
   /**
    * Return all completed missions by the user
-   * @param userId UserId of the volunteer
+   * @param userUid UserId of the volunteer
    */
 
-  async getAllCompletedMissions(userId: string) {
-    const collection = this.getCollection("organizations").doc("1").collection("missions");
-
-    const deliveredMissions = await collection
-      .where("volunteerId", "==", userId)
-      .where("status", "==", MissionStatus.delivered)
-      .get();
-    const succeededMissions = await collection
-      .where("volunteerId", "==", userId)
-      .where("status", "==", MissionStatus.succeeded)
-      .get();
-    const missionsDocumentSnapshot = deliveredMissions.docs.concat(succeededMissions.docs);
-
-    if (missionsDocumentSnapshot.length < 0) {
-      return [];
-    }
-    const missions = missionsDocumentSnapshot.map((doc) => doc.data());
-
-    return missions;
+  getAllCompletedMissions(userUid: string) {
+    return this.getCollection("organizations")
+      .doc(Organization.uid)
+      .collection("missions")
+      .where("volunteerUid", "==", userUid)
+      .where("status", "in", [MissionStatus.succeeded, MissionStatus.failed])
```

Comment:
Did we remove "delivered" from the status? (I think we should).

Reply:
looks like it's still in the schema

Reply:
Then, "Missions Completed" should include missions that have been delivered, cause technically, they are finished and successful.

Reply:
@utunga this is a good question, so missions completed in our book actually mean, a mission is completed only once the mission succeeds or failed, but where does mission delivered go for the volunteer view.

They only have, available, in progress, and completed

Reply:
Yes absolutely we should include both delivered and succeeded in the schema and in the way the app works. It's actually a really important step that a mission has to be not only marked as 'delivered' by the volunteer but actually accepted by the recipient. 

Not only did they drop the box off but the recipient was able to receive and find the box. 

Missions that are completed but not yet succeeded should be in the *IN PROGRESS* tab until they are fully accepted. If the organizer is looking at the In progress tab they should be watching to see that all boxes get not only **delivered** but also **accepted**. They may need to call up recipient or volunteer to work out what went wrong if not. 

In that case, they can override and just mark it as succeeded or failed. This is documented on the [Mission Flow diagram](https://www.figma.com/file/GLg0zPNnwMEGIb9TytCrxQ/Resilience-App?node-id=1257%3A601) _(FWIW I wish I could rename this diagram because its not just for Organisers but everyone)._ 

Its a pretty critical diagram and hasn't changed in a while by the way.



![image](https://user-images.githubusercontent.com/166867/80938718-46760180-8e2e-11ea-8f4d-85d39514e0d8.png)

Reply:
Hmm the swim lanes from top to bottom are recipient, volunteer, recipient.

The white boxes are actions you can undertake.

The gray boxes indicate MissionStatus, and actions between the dotted lines should only happen in that status.